services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy

```diff
@@ -0,0 +1,130 @@
+package org.openkilda.functionaltests.spec.stats
+
+
+import static org.openkilda.functionaltests.extension.tags.Tag.VIRTUAL
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.traffexam.TraffExamService
+import org.openkilda.testing.tools.FlowTrafficExamBuilder
+
+import groovy.time.TimeCategory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Narrative
+import spock.lang.Shared
+
+import javax.inject.Provider
+
+@Narrative("""Now we have two FL instances: Management and Statistics.
+- FL Stats: collect statistics only from the switches.
+- FL Management: do the other work and can collect statistics as well when a switch doesn't connect to FL Stats.""")
+class MflStatSpec extends BaseSpecification {
+    @Shared
+    @Value('${opentsdb.metric.prefix}')
+    String metricPrefix
+
+    @Value('${floodlight.controller.management}')
+    private String managementController
+
+    @Value('${floodlight.controller.stat}')
+    private String statController
+
+    @Autowired
+    Provider<TraffExamService> traffExamProvider
+
+    @Tags(VIRTUAL)
+    def "System is able to collect stats from the statistic and management controllers"() {
+        given: "A flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = 100
+        flowHelper.addFlow(flow)
+
+        when: "Generate traffic on the given flow"
+        def traffExam = traffExamProvider.get()
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildExam(flow, (int) flow.maximumBandwidth)
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stat in openTSDB is created"
+        Date startTime = use(TimeCategory) { new Date() - 20.minutes }
+        def metric = metricPrefix + "flow.raw.bytes"
+        def tags = [switchid: srcSwitch.dpId.toOtsdFormat(), flowid: flow.id]
+        def statsRouterInterval = 100
```

Comment:
brief comment would be good in order to prevent repeating the same investigation you already did on this